datafusion/tests/sql.rs

```diff
@@ -4323,16 +4323,9 @@ async fn test_cast_expressions_error() -> Result<()> {
     let plan = ctx.create_logical_plan(sql).unwrap();
     let plan = ctx.optimize(&plan).unwrap();
     let plan = ctx.create_physical_plan(&plan).unwrap();
-    let result = collect(plan).await;
-
-    match result {
-        Ok(_) => panic!("expected error"),
-        Err(e) => {
-            assert_contains!(e.to_string(),
-                             "Cast error: Cannot cast string 'c' to value of arrow::datatypes::types::Int32Type type"
-            );
-        }
-    }
+    let actual = execute(&mut ctx, sql).await;
+    let expected = vec![vec![""]; 100];
```

Comment:
cast string to Int32 is possible in arrow2, cast failure will result in NULL

https://github.com/jorgecarleitao/arrow2/blob/main/src/compute/cast/mod.rs#L129

Reply:
cool, filed https://github.com/apache/arrow-datafusion/issues/1056